Add exclusions option for logging sink resources

pmoncadaisla opened the pull request:
Added as defined in https://cloud.google.com/logging/docs/reference/v2/rest/v2/projects.sinks#LogExclusion

Log entries that match any of the exclusion filters will not be exported.
If a log entry is matched by both filters and one of exclusion_filters it will not be exported.

Affects to the following resources:

- google_logging_billing_sink
- google_logging_project_sink
- google_logging_organization_sink

slevenick commented on google/resource_logging_sink.go:
```diff
@@ -30,6 +31,38 @@ func resourceLoggingSinkSchema() map[string]*schema.Schema {
+						Description: `An advanced logs filter that matches the log entries to be excluded. By using the sample function, you can exclude less than 100% of the matching log entries`,
+					},
+					"disabled": {
+						Type:        schema.TypeString,
```
Why is this a string but it seems to be a boolean in the API client? We could just use a TypeBool which should remove some of the conversions

pmoncadaisla replied:
Absolutely, there was no reason why `disabled` had to be `TypeString`. I have changed it to `TypeBool`. 
Thanks for the catch!

slevenick reviewed (commented):
Thanks for adding this! I'm running the tests, but is there a reason to have `disabled` be a string rather than a bool?

pmoncadaisla commented:
Having `disabled` as `TypeString` didn't make sense. I have changed it to `TypeBool`.
Thanks for the catch @slevenick !

pmoncadaisla commented on google/resource_logging_project_sink_test.go:
```diff
@@ -248,3 +271,32 @@ resource "google_bigquery_dataset" "logging_sink" {
+  unique_writer_identity = false
+}
+
+resource "google_logging_project_bucket_config" "logbucket" {
```
Sorry for this! I was running tests incorrectly locally and I thought they ran in CI. I have fixed this issue and a couple issues more.

```
stAccLoggingProjectSink_loggingbucket'
==> Checking source code against gofmt...
==> Checking that code complies with gofmt requirements...
go generate  ./...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google -v -run=TestAccLoggingProjectSink_loggingbucket -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google/version.ProviderVersion=acc"
=== RUN   TestAccLoggingProjectSink_loggingbucket
=== PAUSE TestAccLoggingProjectSink_loggingbucket
=== CONT  TestAccLoggingProjectSink_loggingbucket
--- PASS: TestAccLoggingProjectSink_loggingbucket (9.01s)
PASS
ok  	github.com/hashicorp/terraform-provider-google/google	9.955s
```

pmoncadaisla commented:
I have fixed an issue with tests, where unnecessary `google_logging_project_bucket_config` was created wrongly.
Also fixed some old bool conversions and removed `Computed` field from `exclusions` as it does not need to be computed.

slevenick reviewed (commented):
This looks great! Can you add documentation for these fields to the relevant resources?

For example here: https://github.com/hashicorp/terraform-provider-google/blob/master/website/docs/r/logging_project_sink.html.markdown

pmoncadaisla commented:
Awesome, thanks for the catch again.

I have added documentation with an example.

slevenick reviewed (approved):
Thanks for the contribution!
